### PR TITLE
#140 - Alternate config_options style

### PR DESCRIPTION
**Pull request by ccarouge**
Modifications to `config_options.md` in the documentation to improve readability:

- Remove redundant level of heading
- add a link to sub-headings
- replace headings for innermost options with links
- State default values first in the description
- Reorder options to keep the simpler, required options first
- Add code example for every single option
- Add tooltips in examples where useful

### CODE REVIEW

**Comment by SeanBryan51** [comment]
Nice. Can you replace all the options that are headings to 
```
[`option`](#+heading.option){ #+heading.option }
```
I think it would be best if all the options had a consistent look.

**Comment by ccarouge** [comment]
Absolutely. This was just a first try to see what it would look like and if it would improve things before going all the way.

**Comment by ccarouge** [comment]
Actually, I don't think I want all the options to be links instead of headings. I think we want to keep headings for some of them so they appear in the table of contents. My first try is with only the innermost options as "links" and the other levels of options are headings. I was somewhat taking inspiration from [here](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#table-of-contents). It uses headings for all keys that have no values and uses the "link" approach for the sub-keys with values.

Obviously, we have keys at the top level that have values so I'm not sure what will be the best for that case.

**Comment by ccarouge** [comment]
@SeanBryan51 I'm curious to see what you think of it now.

**Comment by SeanBryan51** [comment]
Looks good to me!

**Comment by ccarouge** [comment]
@bschroeter Sean and I have seen this documentation way too many times. I think some fresher eyes for the review would be good.

**Comment by ccarouge** [comment]
@bschroeter you can see a preview of the docs, by clicking on "Details" for the `docs/readthedocs.org:benchcab` test.

File: docs/user_guide/config_options.md [hunk 1]
```diff
@@ -29,107 +29,144 @@ The different running modes of `benchcab` are solely dependent on the options us
...
 
-##### `mem`
+## experiment
 
-: The total memory limit for the PBS job, i.e. the `-l mem=<10GB>` PBS flag (see [PBS Directives Explained][nci-pbs-directives]).
-: This key is **optional** and can be omitted from the config file. By default `mem` is set to `30GB`.
+: **Default:** _required option, no default_. :octicons-dash-24: Type of experiment to run. Experiments are defined in the **benchcab-evaluation** workspace on [modelevaluation.org][meorg]. This key specifies the met forcing files used in the test suite. To choose from:
 
-##### `walltime`
+    - [`forty-two-site-test`][forty-two-me]: to run simulations using 42 FLUXNET sites
```
Comment (bschroeter) [inline docs/user_guide/config_options.md:64]: Is there anything stopping us from putting this in a table?

i.e.

| Key | Simulation site(s) |
|-|-|
|`AU-Tum`| Tumbarumba (AU) |

**Review by bschroeter — COMMENTED** [review]
Looks good. Just a couple of suggestions from me.

1. The admonition used to alert the user to all items being required at the top of the page could be the exclamation point (```!!! warning```). Semantically, an exclamation point usually means "hey, stop, look here!" which is perhaps the message we want to send for that sort of thing. I'm not sure the note icon and colouring conveys the importance and is likely to be overlooked. This does of course change the colour of the box to orange, but perhaps that is a good thing? Note, this isn't actually part of this commit so feel free to ignore this suggestion.

2. It would be good to have a "Putting it all together" section at the end where all of the options just explained are combined into a complete and working configuration as a code block. This will help users who just want to grab a configuration example and drop it in, without having to worry about getting the yaml syntax correct.

**Comment by ccarouge** [comment]
@bschroeter 
> The admonition used to alert the user to all items being required at the top of the page could be the exclamation point (!!! warning).

Actually, I think we can completely remove this. I have now specified for each option whether it is optional or required. So that the look is consistent between options. But I've realised I've used _required option_ but _optional key_. I'll changed all to key.

**Comment by ccarouge** [comment]
@bschroeter About the "put it all together", we already tell users to start from the [`bench_example` repository](https://github.com/CABLE-LSM/bench_example/blob/main/config.yaml). This already contains an example yaml file with all the required options (but not the optional ones). Is that enough? Do you think we should link to it on this doc page? I don't really want to have a copy as they are going to get out of sync.

If people come from the User Guide to this page to see the options, it is linked [from this paragraph]( https://benchcab.readthedocs.io/en/latest/user_guide/#setup-the-work-directory):

> Once the work directory is cloned, you will need to adapt the config.yaml file to your case. Refer to [the description of the options](https://benchcab.readthedocs.io/en/latest/user_guide/config_options/) for this file.

So they should have the example open on their screen.

**Comment by bschroeter** [comment]
> @bschroeter About the "put it all together", we already tell users to start from the [`bench_example` repository](https://github.com/CABLE-LSM/bench_example/blob/main/config.yaml). This already contains an example yaml file with all the required options (but not the optional ones). Is that enough? Do you think we should link to it on this doc page? I don't really want to have a copy as they are going to get out of sync.
> 
> If people come from the User Guide to this page to see the options, it is linked [from this paragraph](https://benchcab.readthedocs.io/en/latest/user_guide/#setup-the-work-directory):
> 
> > Once the work directory is cloned, you will need to adapt the config.yaml file to your case. Refer to [the description of the options](https://benchcab.readthedocs.io/en/latest/user_guide/config_options/) for this file.
> 
> So they should have the example open on their screen.

This is a fair assumption. I can appreciate that the example on the docs will likely get out of date with what is installed.

**Review by bschroeter — APPROVED** [review]

**Pull request merged**